packages/scene-composer/src/utils/entityModelUtils/modelRefComponent.ts

```diff
@@ -0,0 +1,96 @@
+import { ComponentRequest, ComponentUpdateRequest } from '@aws-sdk/client-iottwinmaker';
+import { DocumentType } from '@aws-sdk/types';
+
+import { IModelRefComponent, KnownComponentType } from '../../interfaces';
+import { componentTypeToId } from '../../common/entityModelConstants';
+import { IModelRefComponentInternal } from '../../store';
+import { generateUUID } from '../mathUtils';
+
+enum ModelRefComponentProperty {
+  Uri = 'uri',
+  ModelType = 'modelType',
+  UnitOfMeasure = 'unitOfMeasure',
+  LocalScale = 'localScale',
+  CastShadow = 'castShadow',
+  ReceiveShadow = 'receiveShadow',
+}
+
+export const createModelRefComponent = (model: IModelRefComponent): ComponentRequest => {
+  const comp: ComponentRequest = {
+    componentTypeId: componentTypeToId[KnownComponentType.ModelRef],
+    properties: {},
+  };
+  if (model.uri) {
+    comp.properties![ModelRefComponentProperty.Uri] = {
+      value: {
+        stringValue: model.uri,
+      },
+    };
+  }
+  if (model.modelType) {
+    comp.properties![ModelRefComponentProperty.ModelType] = {
+      value: {
+        stringValue: model.modelType,
+      },
+    };
+  }
+  if (model.unitOfMeasure) {
+    comp.properties![ModelRefComponentProperty.UnitOfMeasure] = {
+      value: {
+        stringValue: model.unitOfMeasure,
+      },
+    };
+  }
+  if (model.localScale) {
+    comp.properties![ModelRefComponentProperty.LocalScale] = {
+      value: {
+        listValue: model.localScale.map((l) => ({ doubleValue: l })),
+      },
+    };
+  }
+  if (model.castShadow) {
+    comp.properties![ModelRefComponentProperty.CastShadow] = {
+      value: {
+        booleanValue: model.castShadow,
+      },
+    };
+  }
+  if (model.receiveShadow) {
+    comp.properties![ModelRefComponentProperty.ReceiveShadow] = {
+      value: {
+        booleanValue: model.receiveShadow,
+      },
+    };
+  }
+  return comp;
+};
+
+export const updateModelRefComponent = (model: IModelRefComponent): ComponentUpdateRequest => {
+  const request = createModelRefComponent(model);
+  return {
+    componentTypeId: request.componentTypeId,
+    propertyUpdates: request.properties,
+  };
+};
+
+export const parseModelRefComp = (model: DocumentType): IModelRefComponentInternal | undefined => {
+  if (!model?.['properties']) {
```

Comment:
you should also check if `uri` and `modelType` exist since we cannot render without them. if for some reason they are not returned from the query, we should return undefined here as we cannot render it